packages/ui/ViewManager/View.js

```diff
@@ -135,13 +135,19 @@ class View extends React.Component {
 		this.animation = null;
 		this._raf = null;
 		this.state = {
-			entering: !props.appearing
+			entering: !props.appearing,
+			changeDirection: false,
+			prevReverseTransition: props.reverseTransition
 		};
 	}
 
-	UNSAFE_componentWillReceiveProps (nextProps) {
-		// changeDirection let's us know we need to switch mid-transition
-		this.changeDirection = this.animation ? this.props.reverseTransition !== nextProps.reverseTransition : false;
+	static getDerivedStateFromProps (props, state) {
+		if (state.prevReverseTransition !== props.reverseTransition) {
+			return {
+				changeDirection: true
```

Comment:
Shouldn't we cache `prevReverseTransition` or was this intentional? Plus, it is understandable that there's a rare case when `changeDirection` needs to be reset to `false` once after set to `true`, but shouldn't we have a reset condition somewhere? This leaves a room for an error IMO.

Reply:
I guess that might happen, but where should I add it?

Reply:
Do you think there's a way to capture mid-transition state?

Reply:
i don't know